framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/resource/impl/ProjectImpl.java

```diff
@@ -100,34 +101,43 @@ public void processTemplateAndCreateResources(URL templateUrl, Map<String, Strin
         }
     }
 
+    static Semaphore semaphore = new Semaphore(1);
```

Comment:
The synchronized on ProjectImpl methods wasn't enough it this case?
What is the source of concurrency here?